openmc/deplete/batchwise.py

```diff
@@ -0,0 +1,1316 @@
+    attrib_name : str
+        Material attribute name
+    mat_vector : dict
+        Dictionary of material composition to parameterize, where a pair key value
+        represents a nuclide and its weight fraction, respectively.
+    bracket : list of float
+        Bracketing interval to search for the solution, always relative to the
+        solution at previous step.
+    bracket_limit : list of float
+        Absolute bracketing interval lower and upper; if during the adaptive
+        algorithm the search_for_keff solution lies off these limits the closest
+        limit will be set as new result.
+    density_treatment : str
+        Whether or not to keep constant volume or density after a depletion step
+        before the next one.
+        Default to 'constant-volume'
+    bracketed_method : {'brentq', 'brenth', 'ridder', 'bisect'}, optional
+        Solution method to use.
+        This is equivalent to the `bracket_method` parameter of the
+        `search_for_keff`.
+        Defaults to 'brentq'.
+    tol : float
+        Tolerance for search_for_keff method.
+        This is equivalent to the `tol` parameter of the `search_for_keff`.
+        Default to 0.01
+    target : Real, optional
+        This is equivalent to the `target` parameter of the `search_for_keff`.
+        Default to 1.0.
+    print_iterations : Bool, Optional
+        Whether or not to print `search_for_keff` iterations.
+        Default to True
+    search_for_keff_output : Bool, Optional
+        Whether or not to print transport iterations during  `search_for_keff`.
+        Default to False
+
+    Attributes
+    ----------
+    material : openmc.Material or int or str
+        OpenMC Material identifier to where apply batch wise scheme
+    mat_vector : dict
+        Dictionary of material composition to parameterize, where a pair key value
+        represents a nuclide and its weight fraction, respectively.
+    attrib_name : str
+        Material attribute name
+
+    """
+
+    def __init__(
+        self,
+        material,
+        attrib_name,
+        operator,
+        mat_vector,
+        bracket,
+        bracket_limit,
+        density_treatment="constant-volume",
+        bracketed_method="brentq",
+        tol=0.01,
+        target=1.0,
+        print_iterations=True,
+        search_for_keff_output=True,
+    ):
+
+        super().__init__(
+            material,
+            operator,
+            mat_vector,
+            bracket,
+            bracket_limit,
+            density_treatment,
+            bracketed_method,
+            tol,
+            target,
+            print_iterations,
+            search_for_keff_output,
+        )
+
+        # Not needed but used for consistency with other classes
+        check_value("attrib_name", attrib_name, "refuel")
+        self.attrib_name = attrib_name
+
+    def _model_builder(self, param):
+        """Callable function which builds a model according to a passed
```

Comment:
Please ensure docstrings have a single line and then an empty line before new content